src/components/Workspace.jsx

```diff
@@ -128,6 +131,24 @@ class Workspace extends React.Component {
     addEventListener('beforeunload', this._confirmUnload);
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
Hey @inlinestyle this is a good thought, but I feel pretty strongly that any interaction with the DOM should occur inside a React component. It would feel like a big muddling of concerns to me for a Saga to be calling `window.open()` etc.

I definitely agree with the sentiment that the window management has nothing to do with `Workspace` semantically. It would feel much better to have a separate `<GistExport>` component whose sole responsibility was to manage the lifecycle of the export window. I initially shied away from that solution because of the browser constraints around opening a `window` based on a user action, although I’m now thinking it might be possible, since there’s no reason to believe there is any asynchronicity in a Redux store update.

Which brings us to…

> comparing with prev props is a minor code smell IMHO

I *kind of* agree. I would say that, if you are comparing with previous props, it’s worth a good hard look at whether there’s a way to change that fact. BUT, there are situations in which it’s unavoidable, particularly for components that have to bridge the gap between React state and imperative APIs that are external to React (e.g. the `Editor` component; the `GistExport` component I propose above)

Generally though I think a component that is watching prop changes probably shouldn’t _also_ going through normal `render` cycles.